src/pybind/mgr/mgr_module.py

```diff
@@ -1214,6 +1244,7 @@ def config_notify(self) -> None:
         """
         pass
 
+    @API.hook
```

Comment:
I don't think methods like `serve` and `shutdown` should be exposed since it is called from ceph-mgr c++ side.